### PR TITLE
Fix the return type of 'jerry_string_substr'.

### PR DESCRIPTION
**Pull request by LaszloLango**
Also added new unit test to test basic functionality of 'jerry_string_substr'.

JerryScript-DCO-1.0-Signed-off-by: Laszlo Lango laszlo.lango@h-lab.eu

### CODE REVIEW

**Review by robertsipka — APPROVED** [review]
LGTM

File: tests/unit-core/test-api-strings.c [hunk 1]
```diff
@@ -159,6 +159,22 @@ main (void)
     jerry_value_free (test_str);
   }
 
+  /* Test jerry_string_substr */
+  {
+    jerry_value_t test_str = jerry_string_sz ("Hello World!");
```
Comment (lygstate) [inline tests/unit-core/test-api-strings.c:164]: Prefer `JERRY_ZSTR_ARG`

Comment (LaszloLango) [reply tests/unit-core/test-api-strings.c]: Its just a test code. It does not make any difference.

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**